web-components/src/components/organisms/CollapseList/CollapseList.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react';
+import { Box, Collapse, SxProps } from '@mui/material';
+
+import { Theme } from '../../../theme/muiTheme';
+import { sxToArray } from '../../../utils/mui/sxToArray';
+import { TextButton } from '../../buttons/TextButton';
+import { LabelSize } from '../../typography/sizing';
+import { SEE_LESS, SEE_MORE } from './CollapseList.constants';
+
+type Props = {
+  items: JSX.Element[];
+  max?: number;
+  buttonTextSize?: LabelSize;
+  sx?: SxProps<Theme>;
+};
+
+export const CollapseList = ({
+  items,
+  max = 3,
+  buttonTextSize = 'xs',
+  sx,
+}: Props): JSX.Element | null => {
+  const [expanded, setExpanded] = useState<boolean>(false);
+
+  if (items.length === 0) {
+    return null;
+  }
+  if (items.length <= max) {
+    return <>{items.map(item => item)}</>;
```

Comment:
No need to change here but I'm not comfortable using preventive `if` in JSX because it makes it easier to break the rules of hooks. One alternative is to generate a unique state based on these conditions and render everything in the tag. Maybe that's worth discussing in the next tech call to decide if want to make it a convention or not.

```
const state = getCollapseState(items); // state = 'empty' | 'below_max' | 'default'
const isBelowMax = state === 'below_max';
const isDefault = state === 'default';

return <>
{isBelowMax && <>{items.map(item => item)}</>}
{isDefault && ...}
</>

```

Reply:
good point, will update